homeassistant/components/smartthings/__init__.py

```diff
@@ -144,31 +144,30 @@ async def retrieve_device_status(device):
         broker.connect()
         hass.data[DOMAIN][DATA_BROKERS][entry.entry_id] = broker
 
+    except APIInvalidGrant as ex:
+        _LOGGER.error(
+            "Unable to obtain a new refresh token for '%s': %s - Reauthorize the SmartApp",
```

Comment:
Isn't it normally that the refresh token is static, and you get new access tokens ?

Reply:
In the ST API, the `refresh token` is single use and only valid for 30 days.  When you call the API for a new token, you get a `access token` and `refresh token` pair.

The integration stores the new refresh token during entry setup, but because the actual persistence is a background task, sometimes the token gets lost if the HA process isn't gracefully shutdown or crashes.  Seems to happen quite a bit for people in the wild.

Reply:
Let's not make it a background task maybe ? If something takes 0.5 second longer (worst case), that's fine.